src/app/components/shared/aoi-options/geocode-selector/geocode-selector.component.ts

```diff
@@ -0,0 +1,77 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { WktService } from '@services';
+import { debounceTime, map, Observable, Subject } from 'rxjs';
+import { Vector as VectorSource } from 'ol/source';
+import GeoJSON from 'ol/format/GeoJSON';
+import { SubSink } from 'subsink';
+import { AppState } from '@store';
+import { Store } from '@ngrx/store';
+import { getGeocodeArea } from '@store/filters';
+
+@Component({
+  selector: 'app-geocode-selector',
+  templateUrl: './geocode-selector.component.html',
+  styleUrls: ['./geocode-selector.component.scss']
+})
+export class GeocodeSelectorComponent implements OnInit, OnDestroy {
+  private key = 'bFwkahiCrAA0526OlsHS';
+  public options = [];
+  public search_key = '';
+  public subject = new Subject();
+  public results$ = new Observable();
+
+
+  private vectorSource;
+
+  private subs = new SubSink();
+  @Output() geocodeWkt = new EventEmitter();
+  constructor(private http: HttpClient,
+    private wkt: WktService,
+    private store$: Store<AppState>
+  ) { }
+
+  ngOnInit(): void {
+    this.results$ = this.subject.pipe(
+      debounceTime(200),
+      map((geocodeText: string) => this.http.get(`https://api.maptiler.com/geocoding/${encodeURIComponent(geocodeText)}.json?key=${this.key}`)));
```

Comment:
Can use `switchMap` here instead of `map` to not have to subscribe twice at line 47